src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -73,42 +74,72 @@ class PeerActor(
 
   def waitingForConnectionResult(rlpxConnection: RLPxConnection, numRetries: Int = 0): Receive =
     handleSubscriptions orElse handleTerminated(rlpxConnection) orElse stashMessages orElse {
-    case RLPxConnectionHandler.ConnectionEstablished =>
-      log.info("RLPx connection established, sending Hello")
-      rlpxConnection.sendMessage(createHelloMsg())
-      val timeout = scheduler.scheduleOnce(3.seconds, self, ProtocolHandshakeTimeout)
-      context become waitingForHello(rlpxConnection, timeout, numRetries)
-
-    case RLPxConnectionHandler.ConnectionFailed =>
-      log.warning("Failed to establish RLPx connection")
-      rlpxConnection.uriOpt match {
-        case Some(uri) if numRetries < peerConfiguration.connectMaxRetries =>
-          context unwatch rlpxConnection.ref
-          scheduleConnectRetry(uri, numRetries)
-        case Some(uri) =>
-          log.warning("No more reconnect attempts left, removing peer")
-          context stop self
-        case None =>
-          log.warning("Connection was initiated by remote peer, not attempting to reconnect")
-          context stop self
-      }
+      case RLPxConnectionHandler.ConnectionEstablished =>
+        processHandshakerNextMessage(initHandshaker, rlpxConnection, numRetries)
+
+      case RLPxConnectionHandler.ConnectionFailed =>
+        log.warning("Failed to establish RLPx connection")
+        rlpxConnection.uriOpt match {
+          case Some(uri) if numRetries < peerConfiguration.connectMaxRetries =>
+            context unwatch rlpxConnection.ref
+            scheduleConnectRetry(uri, numRetries)
+          case Some(_) =>
+            log.warning("No more reconnect attempts left, removing peer")
+            context stop self
+          case None =>
+            log.warning("Connection was initiated by remote peer, not attempting to reconnect")
+            context stop self
+        }
 
-    case GetStatus => sender() ! StatusResponse(Connecting)
-  }
+        case GetStatus => sender() ! StatusResponse(Connecting)
+    }
+
+  def processingHandshaking(handshaker: Handshaker[PeerInfo], rlpxConnection: RLPxConnection,
+                            timeout: Cancellable, numRetries: Int): Receive =
+    handleSubscriptions orElse handleTerminated(rlpxConnection) orElse
+      handleDisconnectMsg orElse handlePingMsg(rlpxConnection) orElse
+      handleBlockFastDownload(rlpxConnection) orElse stashMessages orElse {
```

Comment:
I was wondering, what do you think @LukasGasior1 about incorporating the handling of the response of the fork block on the handshaker? It seems to make more sense as it is part of the handshaking (and also with how the `PeerActor` will be after the network refactoring AFAIK), but we would need something similar to our previously removed `handlePeerChainCheck` https://github.com/input-output-hk/etc-client/commit/4979c7524105391c84de20f1f9296424b535b800#diff-d88dc0e9076bb9b1c58875187ebcd4acL270.